python/lsst/ts/externalscripts/auxtel/latiss_base_align.py

```diff
@@ -0,0 +1,788 @@
+        """Configure script.
+
+        Parameters
+        ----------
+        config : `types.SimpleNamespace`
+            Script configuration, as defined by `schema`.
+
+        Raises
+        ------
+        RuntimeError:
+            If both `find_target` and `track_target` are defined in the
+            configuration.
+        """
+
+        if hasattr(config, "find_target") and hasattr(config, "track_target"):
+            raise RuntimeError(
+                "find_target and track_target configuration sections cannot be specified together."
+            )
+        elif hasattr(config, "find_target"):
+            self.log.debug(f"Finding target for cwfs @ {config.find_target}")
+            self.cwfs_target = await self.atcs.find_target(**config.find_target)
+            self.cwfs_target_ra = None
+            self.cwfs_target_dec = None
+            self.log.debug(f"Using target {self.cwfs_target} for cwfs.")
+        elif hasattr(config, "track_target"):
+            self.log.debug(f"Tracking target {config.track_target} for cwfs.")
+            self.cwfs_target = config.track_target.get("target_name", "cwfs_target")
+            self.cwfs_target_ra = None
+            self.cwfs_target_dec = None
+            if "icrs" in config.track_target:
+                self.cwfs_target_ra = config.track_target["icrs"]["ra"]
+                self.cwfs_target_dec = config.track_target["icrs"]["dec"]
+        else:
+            self.log.debug("No target configured.")
+            self.cwfs_target = None
+            self.cwfs_target_ra = None
+            self.cwfs_target_dec = None
+
+        self.filter = config.filter
+        self.grating = config.grating
+
+        # exposure time for the intra/extra images (in seconds)
+        self.exposure_time = config.exposure_time
+
+        self.acq_exposure_time = config.acq_exposure_time
+
+        # offset for the intra/extra images
+        self.dz = config.dz
+        # delta offset for extra focal image
+        self.extra_focal_offset = config.extra_focal_offset
+
+        self.large_defocus = config.large_defocus
+
+        self.threshold = config.threshold
+
+        self.coma_threshold = config.coma_threshold
+
+        self.offset_telescope = config.offset_telescope
+
+        self.max_iter = config.max_iter
+
+        self.reason = config.reason
+
+        self.program = config.program
+
+        self.take_detection_image = config.take_detection_image
+
+    def set_metadata(self, metadata: salobj.type_hints.BaseMsgType) -> None:
+        """Sets script metadata.
+
+        Parameters
+        ----------
+        metadata : `salobj.type_hints.BaseMsgType`
+            Script metadata topic. The information is set on the topic
+            directly.
+        """
+        # It takes about 300s to run the cwfs code, plus the two acquisition
+        # exposures.
+        metadata.duration = 300.0 + 2.0 * self.exposure_time
```

Comment:
I think we can update this to be max_iterations * exposure_time*2 + detection image + in-focus image time etc. Plus a small fudge factor of course.